File: dotnet/Core/GameX.Resource/Crytek/FarCryNewDawn.cs
using System.Collections.Concurrent;
using System.Collections.Generic;
using System.IO.Compression;
using System.Linq;

namespace GameX.Crytek;

public static class FarCryNewDawn {
    static readonly IDictionary<string, ZipArchiveEntry> Files;
    static FarCryNewDawn() {
        var assembly = typeof(FarCryNewDawn).Assembly;
        var s = assembly.GetManifestResourceStream("GameX.Resource.Crytek.FarCryNewDawn.zip");
        var arc = new ZipArchive(s, ZipArchiveMode.Read);
        Files = arc.Entries.ToDictionary(s => s.FullName);
    }

    static readonly ConcurrentDictionary<string, Dictionary<ulong, string>> FileHashes = new();
    public static Dictionary<ulong, string> GetFileHashes(string path) => FileHashes.GetOrAdd(path, s => Files.TryGetValue(s, out var z) ? FarCryX.HashFilelist64(z) : []);
}

File: python/gamex/Resources/Crytek/FarCryNewDawn.py
from zipfile import ZipFile
from importlib import resources
from .FarCryX import hashFilelist64

f = resources.files().joinpath('FarCryNewDawn.zip').open('rb')
arc: ZipFile = ZipFile(f, 'r')
files: dict[str, object] = { s.filename:s for s in arc.infolist() }
fileHashes: dict[str, dict[int, str]] = {}
@staticmethod
def getFileHashes(path: str) -> dict[int, str]:
    if path in fileHashes: return fileHashes[path]
    fileHashes[path] = hashFilelist64(arc, files[path]) if path in files else []
    return fileHashes[path]
